File: app/src/main/java/com/example/beerrecruitmentexercise/adapter/BeersAdapter.java
package com.example.beerrecruitmentexercise.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.beerrecruitmentexercise.R;
import com.example.beerrecruitmentexercise.dto.BeerDTO;
import com.facebook.drawee.view.SimpleDraweeView;

import java.util.ArrayList;

public class BeersAdapter extends RecyclerView.Adapter<BeersAdapter.BeersViewHolder> {

    private final ArrayList<BeerDTO> beersList;
    private Context context;

    static class BeersViewHolder extends RecyclerView.ViewHolder {

        TextView tvName;
        SimpleDraweeView imageView;

        BeersViewHolder(final View view) {
            super(view);
            tvName = view.findViewById(R.id.tvName);
            imageView = view.findViewById(R.id.imageView);
        }
    }

    public BeersAdapter(final ArrayList<BeerDTO> beers, final Context context) {
        this.beersList = beers;
        this.context = context;
    }

    @NonNull
    @Override
    public BeersViewHolder onCreateViewHolder(final ViewGroup parent, final int viewType) {
        final View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.beer_item, parent, false);
        return new BeersViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull BeersViewHolder holder, int position) {

        holder.tvName.setText(beersList.get(position).getName());
        holder.imageView.setImageURI(beersList.get(position).getImageUrl());
    }

    @Override
    public int getItemCount() {
        return beersList.size();
    }
}

File: app/src/main/java/com/example/beerrecruitmentexercise/adapter/BeersAdapter.kt
package com.example.beerrecruitmentexercise.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.example.beerrecruitmentexercise.R
import com.example.beerrecruitmentexercise.dto.BeerDTO
import com.facebook.drawee.view.SimpleDraweeView

import java.util.ArrayList

class BeersAdapter
/**
 * Constructor for beers adapter
 *
 * @param beers the list of beers to display
 */
(private val beersList: ArrayList<BeerDTO>) : RecyclerView.Adapter<BeersAdapter.BeersViewHolder>() {

    /**
     * Beers viewholder, view to show item in recyclerview
     */
    class BeersViewHolder(view: View) : RecyclerView.ViewHolder(view) {

        var tvName: TextView
        var tvTagline: TextView
        var tvDescription: TextView
        var tvABV: TextView
        var imageView: SimpleDraweeView

        init {
            tvName = view.findViewById(R.id.tvName)
            tvTagline = view.findViewById(R.id.tvTagline)
            tvDescription = view.findViewById(R.id.tvDescription)
            tvABV = view.findViewById(R.id.tvABV)
            imageView = view.findViewById(R.id.imageView)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BeersViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.beer_item, parent, false)
        return BeersViewHolder(view)
    }

    override fun onBindViewHolder(holder: BeersViewHolder, position: Int) {

        holder.tvName.text = beersList[position].name
        holder.tvTagline.text = beersList[position].tagline
        holder.tvDescription.text = beersList[position].description
        holder.tvABV.text = StringBuilder().append("ABV: ").append(beersList[position].abv).append(" %").toString()

        holder.imageView.setImageURI(beersList[position].imageUrl)
    }

    override fun getItemCount(): Int {
        return beersList.size
    }
}
